tools/SetupFlow/DevHome.SetupFlow/Models/RepositoryProvider.cs

```diff
@@ -295,48 +315,67 @@ public IEnumerable<IRepository> GetAllRepositories(IDeveloperId developerId, Dic
+    {
+        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetAllRepos_Event", LogLevel.Critical, new GetReposEvent("CallingExtension", _repositoryProvider.DisplayName, developerId));
+        var repoSearchInformation = new RepositorySearchInformation();
+        try
         {
-            if (_repositories.TryGetValue(developerId, out var _))
+            var result = _repositoryProvider.GetRepositoriesAsync(developerId).AsTask().Result;
+            if (result.Result.Status == ProviderOperationStatus.Success)
             {
-                _repositories[developerId] = result.Repositories;
+                repoSearchInformation.Repositories = result.Repositories;
             }
             else
             {
-                _repositories.Add(developerId, result.Repositories);
+                Log.Logger?.ReportError(Log.Component.RepoConfig, $"Could not get repositories.  Message: {result.Result.DisplayMessage}", result.Result.ExtendedError);
             }
         }
-        else
+        catch (AggregateException aggregateException)
         {
-            if (_repositories.TryGetValue(developerId, out var _))
+            // Because tasks can be canceled DevHome should emit different logs.
+            if (aggregateException.InnerException is OperationCanceledException)
             {
-                _repositories[developerId] = new List<IRepository>();
+                GlobalLog.Logger?.ReportInfo($"Get Repos operation was cancalled.");
             }
             else
             {
-                _repositories.Add(developerId, new List<IRepository>());
+                GlobalLog.Logger?.ReportError($"{aggregateException}");
             }
-
+        }
+        catch (Exception ex)
+        {
+            Log.Logger?.ReportError(Log.Component.RepoConfig, $"Could not get repositories.  Message: {ex}");
         }
 
-        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetAllRepos_Event", LogLevel.Critical, new GetReposEvent("FoundRepos", _repositoryProvider.DisplayName, developerId));
+        SetRepositories(developerId, repoSearchInformation.Repositories);
 
-        return _repositories[developerId];
+        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetAllRepos_Event", LogLevel.Critical, new GetReposEvent("FoundRepos", _repositoryProvider.DisplayName, developerId));
+        return repoSearchInformation;
     }
 
-    /// <summary>
-    /// Gets all the repositories an account has for this provider.
-    /// </summary>
-    /// <param name="developerId">The account to search in.</param>
-    /// <returns>A collection of repositories.  May be empty</returns>
-    public IEnumerable<IRepository> GetAllRepositories(IDeveloperId developerId)
+    private void SetRepositories(IDeveloperId developerId, IEnumerable<IRepository> repos)
     {
-        return GetAllRepositories(developerId, new());
+        if (_repositories.TryGetValue(developerId, out var _))

```

Comment:
Why do you need to check for existence? 
`_repositories[developerId] = repos;`
will add or update KVP if it exists.

Reply:
I thought `[]` on a dictionary throws an exception if the key does not exist.  Turns out I was wrong.  Changed.